docs/proposal/kmesh_support_localityLB.md

```diff
@@ -0,0 +1,154 @@
+
+![locality_lb_pic](pics/locality_lb.svg)
+
+#### Control plane（user-space）
+1. The user-space kmesh process maintains locality, clusterId, and network information, which is updated when xDS receives information about the workload currently on this node. [handleWorkload]
+2. The user-space kmesh also maintains routingPreference, which is updated when xDS receives information about the service currently on this node. [storeServiceData]
+3. The user-space kmesh maintains a map, `backendsByService`, storing all services present within the current kmesh process along with their respective backends, which will participate in the locality rank calculation.
+4. Once both sets of information are maintained, the locality rank is calculated and the locality load balancing policy is updated in the BPF map `KmeshPrio` via the method `[updateLocalityLBPrio]`:
+  - `updateLocalityLBPrio`: Iterates through and compares the locality information of the kmesh process with every backend associated with the service, matching according to the routing rules defined in the service's load balance scope. For example, if the service is configured with a region scope, the region of the kmesh process is compared with the region of the target backend, incrementing the rank value if they match. After comparing each backend, store the UID of that backend in the `UidList` of `PrioValue` corresponding to its rank value and service ID as the key. Each rank corresponds to a set of backends for that priority level.
+  - In `handleWorkload`, when handling workloads with service information, first execute the existing logic `handleDataWithService` to handle the service, then use `storeServiceBackend` to store the backend in `backendsByService`, and finally call `updateLocalityLBPrio`.
+  - In `handleService`, call `updateLocalityLBPrio`.
+  - In `removeWorkloadFromLocalityLBPrio`, perform a full update (since removing a workload requires updating the entire map, as the current `prio map` stores the backend UID list in an array form, making selective deletion of map elements impractical).
+
+#### Data plane（kernel-space）
+When a request is made for a service, the BPF program queries the backend UID list from the prio map using the request's service ID and iterates through the ranks. 
+- In strict mode, it selects backends that match all scopes.
+- In failover mode, it traverses the backend UID array starting from the highest rank and decrementing step by step. 
+- From the queried backend UIDs, a backend is randomly selected to serve the request.
+
+
+#### data struct
+1. prio map
+
+workload.h
+```
+typedef struct {
+    __u32 service_id; // service id
+    __u32 rank; // rank
+} prio_key;
+typedef struct {
+    __u32 count; // count of current prio
+    __u32 uid_list[MAP_SIZE_OF_PRIO]; // workload_uid to backend
```

Comment:
This can waste memory

So why not add priority to endpoint key

update endpoint_key {

typedef struct {
    __u32 service_id;    // service id
    __u32 Priority,
    __u32 backend_index; // if endpoint_count = 3, then backend_index = 0/1/2
} endpoint_key;